src/storm/modelchecker/helper/conditional/ConditionalHelper.cpp

```diff
@@ -0,0 +1,772 @@
+        STORM_LOG_DEBUG("Initial state is terminal.");
+        if (normalForm.conditionStates.get(initialState)) {
+            return normalForm.targetValue(initialState);  // The value is already known, nothing to do.
+        } else {
+            STORM_LOG_THROW(!normalForm.universalObservationFailureStates.get(initialState), storm::exceptions::NotSupportedException,
+                            "Trying to compute undefined conditional probability: the condition has probability 0 under all policies.");
+            // The last case for a terminal initial state is that it is already target and the condition is reachable with non-zero probability.
+            // In this case, all schedulers induce a conditional probability of 1 (or do not reach the condition, i.e., have undefined value)
+            return storm::utility::one<SolutionType>();
+        }
+    } else {
+        // Catch the case where all terminal states have value zero
+        if (normalForm.nonZeroTargetStateValues.empty()) {
+            return storm::utility::zero<SolutionType>();
+        };
+    }
+    return std::nullopt;  // No trivial case applies, we need to compute the value.
+}
+
+}  // namespace internal
+
+template<typename ValueType, typename SolutionType>
+std::unique_ptr<CheckResult> computeConditionalProbabilities(Environment const& env, storm::solver::SolveGoal<ValueType, SolutionType>&& goal,
+                                                             storm::storage::SparseMatrix<ValueType> const& transitionMatrix,
+                                                             storm::storage::SparseMatrix<ValueType> const& backwardTransitions,
+                                                             storm::storage::BitVector const& targetStates, storm::storage::BitVector const& conditionStates) {
+    // We might require adapting the precision of the solver to counter error propagation (e.g. when computing the normal form).
+    auto normalFormConstructionEnv = env;
+    auto analysisEnv = env;
+    if (env.solver().isForceSoundness()) {
+        normalFormConstructionEnv.solver().minMax().setPrecision(env.solver().minMax().getPrecision() *
+                                                                 storm::utility::convertNumber<storm::RationalNumber, std::string>("1/10"));
+        analysisEnv.solver().minMax().setPrecision(env.solver().minMax().getPrecision() *
+                                                   storm::utility::convertNumber<storm::RationalNumber, std::string>("9/10"));
+    }
+
+    // We first translate the problem into a normal form.
+    // @see doi.org/10.1007/978-3-642-54862-8_43
+    STORM_LOG_THROW(goal.hasRelevantValues(), storm::exceptions::NotSupportedException,
+                    "No initial state given. Conditional probabilities can only be computed for models with a single initial state.");
+    STORM_LOG_THROW(goal.relevantValues().getNumberOfSetBits() == 1, storm::exceptions::NotSupportedException,
```

Comment:
Side comment. I guess we check often whether the number of setBits is exaclty one, maybe we should have a method for that?